src/azure-cli/azure/cli/command_modules/vm/_validators.py

```diff
@@ -271,9 +275,15 @@ def _parse_image_argument(cmd, namespace):
                                            'images', 'Microsoft.Compute')
         return 'image_id'
     except CloudError:
-        err = 'Invalid image "{}". Use a valid image URN, custom image name, custom image id, VHD blob URI, or ' \
-              'pick an image from {}.\nSee vm create -h for more information on specifying an image.'
-        raise CLIError(err.format(namespace.image, [x['urnAlias'] for x in images]))
+        if images is not None:
+            err = 'Invalid image "{}". Use a valid image URN, custom image name, custom image id, ' \
+                  'VHD blob URI, or pick an image from {}.\nSee vm create -h for more information ' \
+                  'on specifying an image.'.format(namespace.image, [x['urnAlias'] for x in images])
+        else:
+            err = 'Failed to connect to remote source of image aliases. Invalid image "{}". Use a ' \
+                  'valid image URN, custom image name, custom image id, or VHD blob URI.\nSee vm ' \
+                  'create -h for more information on specifying an image.'.format(namespace.image)
```

Comment:
If it's not a valid alias nor an existing image name, what will happen?

Reply:
If it's not a valid alias then it goes to `#5` as before, if it's not the name of an existing managed image it raises the same error message as before.